src/composables/useGetParticipants.js

```diff
@@ -0,0 +1,147 @@
+
+	// Exposed
+	const participantsInitialised = ref(false)
+
+	/**
+	 * Initialise the get participants functionality
+	 *
+	 */
+	function initialiseGetParticipants() {
+		EventBus.$on('route-change', onRouteChange)
+		EventBus.$on('joined-conversation', onJoinedConversation)
+
+		// FIXME this works only temporary until signaling is fixed to be only on the calls
+		// Then we have to search for another solution. Maybe the room list which we update
+		// periodically gets a hash of all online sessions?
+		EventBus.$on('signaling-participant-list-changed', debounceUpdateParticipants)
+		subscribe('guest-promoted', onJoinedConversation)
+	}
+
+	/**
+	 * Stop the get participants functionality
+	 *
+	 */
+	function stopGetParticipants() {
+		EventBus.$off('route-change', onRouteChange)
+		EventBus.$off('joined-conversation', onJoinedConversation)
+		EventBus.$off('signaling-participant-list-changed', debounceUpdateParticipants)
+		unsubscribe('guest-promoted', onJoinedConversation)
+	}
+
+	const onRouteChange = () => {
+		participantsInitialised.value = store.getters.participantsList(token.value).length > 1
+	}
+
+	const onJoinedConversation = () => {
+		if (isOneToOneConversation.value) {
+			cancelableGetParticipants()
+		} else {
+			nextTick(debounceUpdateParticipants())
+		}
+	}
+
+	const debounceUpdateParticipants = () => {
+		if (!isActive.value && !isInCall.value) {
+			// Update is ignored but there is a flag to force the participants update
+			pendingChanges = true
+			return
+		}
+
+		if (store.getters.windowIsVisible() && (isInCall.value || !conversation.value?.hasCall)) {
+			debounceFastUpdateParticipants()
+		} else {
+			debounceSlowUpdateParticipants()
+		}
+	    pendingChanges = false
+	}
+
+	const cancelableGetParticipants = async () => {
+		const isInLobby = store.getters.isInLobby
+		const isModeratorOrUser = store.getters.isModeratorOrUser
+		if (fetchingParticipants || token.value === '' || isInLobby || !isModeratorOrUser) {
+			return
+		}
+
+		fetchingParticipants = true
+		debounceFastUpdateParticipants.clear()
+		debounceSlowUpdateParticipants.clear()
+
+		const response = await store.dispatch('fetchParticipants', { token: token.value })
+		if (response) {
+			participantsInitialised.value = true
+		}
+		fetchingParticipants = false
+	}
+
+	const debounceFastUpdateParticipants = debounce(
+		cancelableGetParticipants, 3000)
+	const debounceSlowUpdateParticipants = debounce(
+		cancelableGetParticipants, 15000)
+
+	// Group conversations have composable in RightSidebar, so should work only for one-to-one
+	watch(isOneToOneConversation, (newValue) => {
+		if (newValue) {
+			initialiseGetParticipants()
+		} else {
+			stopGetParticipants()
+		}
+	}, { immediate: true })
+
+	watch(isActive, (newValue) => {
+		if (newValue && pendingChanges) {
+			debounceUpdateParticipants()
+		}
+	})
+
+	return {
+		participantsInitialised,
+		initialiseGetParticipants,
+		stopGetParticipants,
+		cancelableGetParticipants,
+	}
```

Comment:
Nice to move `watch` here. It actually makes `initialiseGetParticipants` and `stopGetParticipants` unneeded outside composable, making it even simpler to use.

Only `beforeDestroy` from `ParticipantsTab.vue` should also be moved to composable.